src/bootstrap/test.rs

```diff
@@ -664,16 +688,16 @@ impl Step for Clippy {
             &[],
         );
 
-        // clippy tests need to know about the stage sysroot
-        cargo.env("SYSROOT", builder.sysroot(compiler));
```

Comment:
Is ``SYSROOT`` not required anymore to run Clippy? 🤔

Reply:
It was moved to `tool.rs` to avoid rebuilding so much; theoretically this PR should probably be based off of https://github.com/rust-lang/rust/pull/96757, but I wanted to avoid stacking too many PRs on top of each other.